wifiphisher/interfaces.py

```diff
@@ -0,0 +1,440 @@
+"""
+This module was made to handle all the interface related operations for
+Wifiphisher.py
+"""
+
+import subprocess
+import os
+import re
+
+DEVNULL = open(os.devnull, "w")
+
+
+class NotEnoughInterfacesFound(Exception):
+    """
+    Exception class to raise in case of a finding less than enough interfaces.
+    """
+
+    def __init__(self):
+        Exception.__init__(self,
+                           "Not enough interfaces were found! Try again.")
+
+class SetMonitorModeError(Exception):
+    """
+    Exception class to raise in case of catching any errors in
+    set_interface_mode function.
+    """
+
+    def __init__(self, error_message):
+        Exception.__init__(self,
+                           ("An error has detected in set_interface_mode.\n" +
+                            error_message))
+
+
+class NoApInterfaceFound(Exception):
+    """
+    Exception class to raise in case of a not finding a valid AP interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "No interface with AP mode was found!")
+
+
+class NoMonitorInterfaceFound(Exception):
+    """
+    Exception class to raise in case of a not finding a valid monitor
+    interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "No interface with monitor mode was found!")
+
+
+class JammingInterfaceInvalid(Exception):
+    """
+    Exception class to raise in case of a invalid jamming interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "Invalid jamming interface (-jI) was given!")
+
+
+class ApInterfaceInvalid(Exception):
+    """
+    Exception class to raise in case of a invalid ap interface.
+    """
+
+    def __init__(self):
+        Exception.__init__(self, "Invalid AP interface (-aI) was given!")
+
+
+def check_compatibility(interface):
+    """
+    Check the compatibility of the interface in regards to both monitor
+    mode and AP mode.
+
+    Args:
+        interface (str): An interface to be checked.
+
+    Returns:
+        ((bool), (bool)): The first item indicates if the interface has monitor
+                          mode and the second item indicates if AP mode is
+                          available.
+
+    Dependencies:
+        word_in_sentence
+    """
+
+    # Initialize variable names
+    device_name = str()
+    has_monitor = bool()
+    has_ap = bool()
```

Comment:
I understand that in python there is no variable initialization. It's more of a declaration rather than initialization and I placed them to make the code clearer.